aptos-move/framework/src/natives/transaction_context.rs

```diff
@@ -112,6 +137,253 @@ fn native_get_script_hash(
     )])
 }
 
+fn native_sender_internal(
+    context: &mut SafeNativeContext,
+    mut _ty_args: Vec<Type>,
+    _args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    context.charge(TRANSACTION_CONTEXT_SENDER_BASE)?;
+
+    let user_transaction_context_opt = get_user_transaction_context_opt_from_context(context);
+    if let Some(transaction_context) = user_transaction_context_opt {
+        Ok(smallvec![Value::address(transaction_context.sender())])
+    } else {
+        Err(SafeNativeError::Abort {
+            abort_code: error::invalid_state(abort_codes::ETRANSACTION_CONTEXT_NOT_AVAILABLE),
+        })
+    }
+}
+
+fn native_secondary_signers_internal(
+    context: &mut SafeNativeContext,
+    mut _ty_args: Vec<Type>,
+    _args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    context.charge(TRANSACTION_CONTEXT_SECONDARY_SIGNERS_BASE)?;
+
+    let user_transaction_context_opt = get_user_transaction_context_opt_from_context(context);
+    if let Some(transaction_context) = user_transaction_context_opt {
+        let secondary_signers = transaction_context.secondary_signers();
+        context.charge(
+            TRANSACTION_CONTEXT_SECONDARY_SIGNERS_PER_SIGNER
+                * NumArgs::new(secondary_signers.len() as u64),
+        )?;
+        Ok(smallvec![Value::vector_address(secondary_signers)])
+    } else {
+        Err(SafeNativeError::Abort {
+            abort_code: error::invalid_state(abort_codes::ETRANSACTION_CONTEXT_NOT_AVAILABLE),
+        })
+    }
+}
+
+fn native_gas_payer_internal(
+    context: &mut SafeNativeContext,
+    mut _ty_args: Vec<Type>,
```

Comment:
We do not need this `mut` here and in other natives?

Reply:
Good catch! Done removing them.